mistralrs-core/src/engine/mod.rs

```diff
@@ -215,19 +215,36 @@ impl Engine {
             None
         };
 
-        let res = handle_seq_error!(
-            pipeline
+        let text = match reason {
```

Comment:
The use of `get_toks` here was and is incorrect when a prefix cache item is being used (https://github.com/EricLBuehler/mistral.rs/blob/master/mistralrs-core/src/sequence.rs#L204-L209)

I could not execute it because the prefix cache crashes https://github.com/EricLBuehler/mistral.rs/issues/121

I'm trying to understand the prefix cache code to propose a fix. Ideally the fix would add support for caching the other way around, where the cache has a subset of a prompt.

Reply:
Actually, that is intentional and only used if we used a cache which had a prompt that was a superset of our prompt. In that case `get_toks` returns the remaining tokens. I don't think this problem affects this PR?

Reply:
I think it does... I don't understand how `get_toks` would return the decoded tokens of the response... They contain the tokens of the prompt, no?

Reply:
No, it returns the tokens which were not in the cached prompt, please see PrefixCacheManager::search_for_matching_cache, specifically the cases where it returns Subset. The toks which are in that variant are returned by `get_toks` for the prompt pass.